### PR TITLE
[6.4x] RHBPMS-4184,JBPM-5277 - getTasksAssignedAsPotentialOwner doesn't allow groups searching, only actorid

### PR DESCRIPTION
**Pull request by mrietveld**
Related to the release this week. 

PR Set: 
- https://github.com/droolsjbpm/droolsjbpm-knowledge/pull/160
- https://github.com/droolsjbpm/jbpm/pull/560
- https://github.com/droolsjbpm/droolsjbpm-integration/pull/605

### CODE REVIEW

**Comment by mrietveld** [comment]
@psiroky Would you mind reviewing and merging (all 3 PRs, of course)? Thanks!

File: kie-internal/src/main/java/org/kie/internal/task/api/TaskQueryService.java [hunk 1]
```diff
@@ -29,102 +29,104 @@
...
+
     /**
-     * 
+     *
      * @param queryWhere
      * @return
      */
     List<TaskSummary> query( String userId, Object queryWhere );
+
+    List<TaskSummary> getTasksByPotentialOwnerGroups( List<String> groupIds );
```
Comment (psiroky) [inline kie-internal/src/main/java/org/kie/internal/task/api/TaskQueryService.java]: This method uses plural (groups)and the other method above (`getTasksByGroup`) uses singular (group). Is that on purpose? (e.g. due to being consistent on within the different interfaces, even though those are consistent between itself)

File: kie-internal/src/main/java/org/kie/internal/task/api/InternalTaskService.java [hunk 2]
```diff
@@ -39,32 +38,24 @@
...
+
     void addGroup(Group group);
 
     void addUser(User user);
 
     int archiveTasks(List<TaskSummary> tasks);
 
-    void claim(long taskId, String userId, List<String> groupIds);
-
-    void claimNextAvailable(String userId, List<String> groupIds);
```
Comment (psiroky) [inline kie-internal/src/main/java/org/kie/internal/task/api/InternalTaskService.java]: What was the reason to remove these methods? (sorry if they were just moved, but I could not find where...)

Comment (mrietveld) [inline kie-internal/src/main/java/org/kie/internal/task/api/InternalTaskService.java]: 2 reasons: 
1. Not being used
2. They weren't even implemented (correctly). We don't have code that can `claim` or `claimNextAvailable` based on group ids. The implementation code was just doing the same thing as the normal `claim` and `claimNextAvailable` methods.

Comment (psiroky) [inline kie-internal/src/main/java/org/kie/internal/task/api/InternalTaskService.java]: Thanks for the explanation!

**Comment by psiroky** [comment]
+1

**Pull request merged**